addon/lib/metrics.js

```diff
@@ -121,13 +151,32 @@ module.exports = {
     this.updateExperiment(addonId, {features: features});
   },
 
+  onReceiveVariantDefs: function(ev) {
+    if (!store.experimentVariants) {
+      store.experimentVariants = {};
+    }
+
+    const { subject, data } = ev;
+    const dataParsed = variantMaker.parseTests(JSON.parse(data));
+
+    store.experimentVariants[subject] = dataParsed;
+    Events.emit(EVENT_SEND_VARIANTS, {
+      data: JSON.stringify(dataParsed),
+      subject: self.id
+    });
+  },
+
   onExperimentPing: function(ev) {
     const { subject, data } = ev;
     const dataParsed = JSON.parse(data);
 
     // TODO: Map add-on ID (subject) to other pingTypes as necessary
     const pingType = 'testpilottest';
 
+    if (subject in store.experimentVariants) {
```

Comment:
Actually, I misread: I thought this was in the payload envelope (down on line 180) and not in the `dataParsed`... the envelope is actually a better place for that variant data, if we're adding it to the `testpilottest` ping.